bundles/org.openhab.binding.anel/src/main/java/org/openhab/binding/anel/internal/discovery/AnelDiscoveryService.java

```diff
@@ -0,0 +1,221 @@
+                            receivePort++;
+                        }
+                    }
+                }
+            } catch (Exception e) {
+
+                logger.warn("Unexpected exception during anel device scan", e);
+
+            } finally {
+
+                scanRunning = false;
+            }
+        }
+        logger.debug("Scan finished.");
+    }
+
+    /* @return Whether or not a device was found for the given broadcast address and port. */
+    private boolean scan(String broadcastAddress, int sendPort, int receivePort) throws IOException {
+        if (scanRunning) {
+            logger.debug("Scanning {}:{}...", broadcastAddress, sendPort);
+            final AnelUdpConnector udpConnector = new AnelUdpConnector(broadcastAddress, receivePort, sendPort,
+                    scheduler);
+
+            try {
+                final boolean[] deviceDiscovered = new boolean[] { false };
+                udpConnector.connect(status -> {
+                    // avoid the same device to be discovered multiple times for multiple responses
+                    if (!deviceDiscovered[0]) {
+                        boolean discoverDevice = true;
+                        synchronized (this) {
+                            if (deviceDiscovered[0]) {
+                                discoverDevice = false; // already discovered by another thread
+                            } else {
+                                deviceDiscovered[0] = true; // we discover the device!
+                            }
+                        }
+                        if (discoverDevice) {
+                            // discover device outside synchronized-block
+                            deviceDiscovered(status, sendPort, receivePort);
+                        }
+                    }
+                }, false);
+
+                udpConnector.send(IAnelConstants.BROADCAST_DISCOVERY_MSG);
+
+                // answer expected within 50-600ms on a regular network; wait up to 2sec just to make sure
+                for (int delay = 0; delay < 10 && !deviceDiscovered[0]; delay++) {
+                    Thread.sleep(100 * DISCOVER_DEVICE_TIMEOUT_SECONDS); // wait 10 x 200ms = 2sec
+                }
+
+                return deviceDiscovered[0];
+
+            } catch (BindException e) {
+
+                // most likely socket is already in use, ignore this exception.
+                logger.debug(
+                        "Invalid address {} or one of the ports {} or {} is already in use. Skipping scan of these ports.",
+                        broadcastAddress, sendPort, receivePort);
+
+            } catch (InterruptedException e) {
+
+                // interrupted...
+
+            } finally {
+
+                udpConnector.disconnect();
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        scanRunning = false;
+        super.stopScan();
```

Comment:
Why not using the thread's interrupt flag via `thread.interrupt()`, which is exactly for this purpose?